### PR TITLE
Merging keymaps for user profile PeterFalken.

### PR DESCRIPTION
**Pull request by PeterFalken**
## Description

Merging keymaps for PeterFalken.
Key boards:
Adafruit: Macropad
OLKB: Planck/rev6_drop & Preonic/rev3_drop
System76: Launch_1
Work Louder: loop, micro, nano & work_board

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/adafruit/macropad/keymaps/PeterFalken/rules.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/adafruit/macropad/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/planck/keymaps/PeterFalken/config.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#define RGBLIGHT_DISABLE_KEYCODES
+#define RGBLIGHT_DEFAULT_VAL 60
+
+// Space optimizations - 01
+#undef LOCKING_SUPPORT_ENABLE
+#undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 02
+#undef RGBLIGHT_ANIMATIONS
+
```
Comment (waffle87) [inline keyboards/planck/keymaps/PeterFalken/config.h:42]:
```suggestion
```

File: keyboards/planck/keymaps/PeterFalken/readme.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# The Default Planck Layout
```
Comment (waffle87) [inline keyboards/planck/keymaps/PeterFalken/readme.md:1]: This file should be removed.

File: keyboards/planck/keymaps/PeterFalken/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/planck/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/preonic/keymaps/PeterFalken/config.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#define RGBLIGHT_DISABLE_KEYCODES
+#define RGBLIGHT_DEFAULT_VAL 60
+
+// Space optimizations - 01
+#undef LOCKING_SUPPORT_ENABLE
+#undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 02
+#undef RGBLIGHT_ANIMATIONS
+
```
Comment (waffle87) [inline keyboards/preonic/keymaps/PeterFalken/config.h:42]:
```suggestion
```

File: keyboards/preonic/keymaps/PeterFalken/readme.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# The default Preonic layout - largely based on the Planck's
```
Comment (waffle87) [inline keyboards/preonic/keymaps/PeterFalken/readme.md:1]: This file should be removed.

File: keyboards/preonic/keymaps/PeterFalken/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/preonic/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/system76/launch_1/keymaps/PeterFalken/config.h [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    #undef RGB_DISABLE_TIMEOUT
+    #undef RGB_MATRIX_MAXIMUM_BRIGHTNESS
+    #undef RGB_MATRIX_DEFAULT_MODE
+    #undef RGB_MATRIX_DEFAULT_HUE
+    #undef RGB_MATRIX_DEFAULT_SAT
+    #undef RGB_MATRIX_DEFAULT_VAL
+
+    // Redefine values
+    // Number of milliseconds to wait until RGB automatically turns off
+    #define RGB_DISABLE_TIMEOUT 5000
```
Comment (waffle87) [inline keyboards/system76/launch_1/keymaps/PeterFalken/config.h:50]:
```suggestion
    #define RGB_MATRIX_TIMEOUT 5000
```

File: keyboards/system76/launch_1/keymaps/PeterFalken/rules.mk [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/system76/launch_1/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/work_louder/loop/keymaps/PeterFalken/config.h [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+#undef RGB_MATRIX_DEFAULT_MODE
+#define RGB_MATRIX_DEFAULT_MODE 0
+
+// Space optimizations - 01
+#undef LOCKING_SUPPORT_ENABLE
+#undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 02
+#undef RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/work_louder/loop/keymaps/PeterFalken/config.h:26]:
```suggestion
#undef RGBLIGHT_EFFECT_BREATHING
#undef RGBLIGHT_EFFECT_RAINBOW_MOOD
#undef RGBLIGHT_EFFECT_RAINBOW_SWIRL
#undef RGBLIGHT_EFFECT_SNAKE
#undef RGBLIGHT_EFFECT_KNIGHT
#undef RGBLIGHT_EFFECT_CHRISTMAS
#undef RGBLIGHT_EFFECT_STATIC_GRADIENT
#undef RGBLIGHT_EFFECT_RGB_TEST
#undef RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/work_louder/loop/keymaps/PeterFalken/config.h [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+/* disable debug print */
+// #define NO_DEBUG
+
+/* disable print */
+// #define NO_PRINT
+
+// #define WORK_LOUDER_LED_PIN_1 B7
+// #define WORK_LOUDER_LED_PIN_2 B6
+// #define WORK_LOUDER_LED_PIN_3 B5
```
Comment (waffle87) [inline keyboards/work_louder/loop/keymaps/PeterFalken/config.h:66]:
```suggestion
```

File: keyboards/work_louder/loop/keymaps/PeterFalken/readme.md [hunk 12]
```diff
@@ -0,0 +1 @@
+# The default keymap for loop
```
Comment (waffle87) [inline keyboards/work_louder/loop/keymaps/PeterFalken/readme.md:1]: This file should be removed.

File: keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+RGBLIGHT_ENABLE = no	# Disable RGB underglow
+RGB_MATRIX_ENABLE = yes	# Disable RGB key matrix
+SPACE_CADET_ENABLE = no	# Free up space
+
+# Enable Link Time Optimizations
+LTO_ENABLE = yes
+
+SRC += rgb_functions.c
+
+DEFAULT_FOLDER = work_louder/loop/rev3
```
Comment (waffle87) [inline keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk:23]:
```suggestion
```
Already at keyboard level.

File: keyboards/work_louder/micro/keymaps/PeterFalken/config.h [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+#ifndef USB_SUSPEND_WAKEUP_DELAY
+    #define USB_SUSPEND_WAKEUP_DELAY 200
+#endif
+
+// Space optimizations - 01
+#undef LOCKING_SUPPORT_ENABLE
+#undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 02
+#undef RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/work_louder/micro/keymaps/PeterFalken/config.h:23]:
```suggestion

// Space optimizations - 01
#undef LOCKING_SUPPORT_ENABLE
#undef LOCKING_RESYNC_ENABLE

// Space optimizations - 02
#undef RGBLIGHT_EFFECT_BREATHING
#undef RGBLIGHT_EFFECT_RAINBOW_MOOD
#undef RGBLIGHT_EFFECT_RAINBOW_SWIRL
#undef RGBLIGHT_EFFECT_SNAKE
#undef RGBLIGHT_EFFECT_KNIGHT
#undef RGBLIGHT_EFFECT_CHRISTMAS
#undef RGBLIGHT_EFFECT_STATIC_GRADIENT
#undef RGBLIGHT_EFFECT_RGB_TEST
#undef RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/work_louder/micro/keymaps/PeterFalken/rules.mk [hunk 16]
```diff
@@ -0,0 +1,19 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/work_louder/micro/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/work_louder/nano/keymaps/PeterFalken/readme.md [hunk 17]
```diff
@@ -0,0 +1 @@
+# The default keymap for nano
```
Comment (waffle87) [inline keyboards/work_louder/nano/keymaps/PeterFalken/readme.md:1]: This file should be removed.

File: keyboards/work_louder/nano/keymaps/PeterFalken/rules.mk [hunk 18]
```diff
@@ -0,0 +1,20 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/work_louder/nano/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

File: keyboards/work_louder/work_board/keymaps/PeterFalken/config.h [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+ // Space optimizations - 01
+ #undef LOCKING_SUPPORT_ENABLE
+ #undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 01
+#undef LOCKING_SUPPORT_ENABLE
+#undef LOCKING_RESYNC_ENABLE
+
+// Space optimizations - 02
+#undef RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/work_louder/work_board/keymaps/PeterFalken/config.h:44]:
```suggestion
#undef RGBLIGHT_EFFECT_BREATHING
#undef RGBLIGHT_EFFECT_RAINBOW_MOOD
#undef RGBLIGHT_EFFECT_RAINBOW_SWIRL
#undef RGBLIGHT_EFFECT_SNAKE
#undef RGBLIGHT_EFFECT_KNIGHT
#undef RGBLIGHT_EFFECT_CHRISTMAS
#undef RGBLIGHT_EFFECT_STATIC_GRADIENT
#undef RGBLIGHT_EFFECT_RGB_TEST
#undef RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/work_louder/work_board/keymaps/PeterFalken/keymap.c [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+  _LOWER,
+  _RAISE,
+  _ADJUST
+};
+
+// Custom keycode definitions
+enum custom_keycodes {
+  KC_LAUNCHPAD = SAFE_RANGE
+};
+
```
Comment (waffle87) [inline keyboards/work_louder/work_board/keymaps/PeterFalken/keymap.c:30]:
```suggestion
```

File: keyboards/work_louder/work_board/keymaps/PeterFalken/readme.md [hunk 21]
```diff
@@ -0,0 +1 @@
+# The default keymap for work_board
```
Comment (waffle87) [inline keyboards/work_louder/work_board/keymaps/PeterFalken/readme.md:1]: This file should be removed.

File: keyboards/work_louder/work_board/keymaps/PeterFalken/rules.mk [hunk 22]
```diff
@@ -0,0 +1,20 @@
+# Enable VIAL support when merged into QMK
+# ENABLE_VIAL = yes
+
```
Comment (waffle87) [inline keyboards/work_louder/work_board/keymaps/PeterFalken/rules.mk:3]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/work_louder/work_board/keymaps/PeterFalken/keymap.c (hunk 20 above)
Comment (PeterFalken) [reply keyboards/work_louder/work_board/keymaps/PeterFalken/keymap.c:30]: Added the custom key handling function for this, but will update the keymap.c file later.

File: keyboards/system76/launch_1/keymaps/PeterFalken/config.h (hunk 8 above)
Comment (PeterFalken) [reply keyboards/system76/launch_1/keymaps/PeterFalken/config.h]: Thanks @waffle87, I missed this one.

File: keyboards/work_louder/loop/keymaps/PeterFalken/config.h (hunk 10 above)
Comment (PeterFalken) [reply keyboards/work_louder/loop/keymaps/PeterFalken/config.h]: No extra space was gained from undefining the RGBLIGHT_EFFECT_*

File: keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk (hunk 14 above)
Comment (PeterFalken) [reply keyboards/work_louder/loop/keymaps/PeterFalken/rules.mk]: I just left the LTO_ENABLE line

**Comment by PeterFalken** [comment]
I hope there was a way to disable/undefine all RGB_MATRIX effects similar to
`#undef RGBLIGHT_ANIMATIONS`

File: keyboards/work_louder/micro/keymaps/PeterFalken/config.h [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+// #define NO_DEBUG
+
+/* disable print */
+// #define NO_PRINT
+
+#undef WORK_LOUDER_LED_PIN_1
+#undef WORK_LOUDER_LED_PIN_2
+#undef WORK_LOUDER_LED_PIN_3
+
+#define WORK_LOUDER_LED_PIN_1 B5
```
Comment (drashna) [inline keyboards/work_louder/micro/keymaps/PeterFalken/config.h:60]: If you're going left to right, this is reverts.

**Review by drashna — COMMENTED** [review]
Please rename the folders to be all lower case.

File: keyboards/adafruit/macropad/keymaps/PeterFalken/keymap.c [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+  [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+};
+#endif
+
+
+#ifdef OLED_ENABLE
+static void render_qmk_logo(void) {
+  static const char PROGMEM qmk_logo[] = {
```
[the remark this answers is not in the document]
Comment (PeterFalken) [reply keyboards/adafruit/macropad/keymaps/PeterFalken/keymap.c]: Yes, this comes from the default profile.
But will need to investigate how to best make use of it.

File: keyboards/work_louder/micro/keymaps/PeterFalken/config.h (hunk 23 above)
Comment (PeterFalken) [reply keyboards/work_louder/micro/keymaps/PeterFalken/config.h]: It seems that the default mapping has the LEDs in order 2, 1, 3

**Review by drashna — APPROVED** [review]

File: keyboards/preonic/keymaps/peterfalken/config.h [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/peterfalken/config.h:64]:
```suggestion
```

File: keyboards/planck/keymaps/peterfalken/config.h [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/peterfalken/config.h:64]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by PeterFalken** [comment]
@fauxpark, I think the changes you requested should be good now.

File: keyboards/preonic/keymaps/peterfalken/keymap.c [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+//                 muse_mode = true;
+//             } else {
+//                 muse_mode = false;
+//             }
+//     }
+//     return true;
+// }
+
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/preonic/keymaps/peterfalken/keymap.c:185]:
```suggestion
void housekeeping_task_user(void) {
```

**Review by drashna — COMMENTED** [review]
You may want to consider using the userspace feature, since you have a lot of repeated code, here.

https://docs.qmk.fm/#/feature_userspace

**Review by drashna — APPROVED** [review]

**Comment by PeterFalken** [comment]
Replaced by [PR-19897](https://github.com/qmk/qmk_firmware/pull/19897)

**Pull request closed without merging**